### PR TITLE
feat: added async function action

### PR DESCRIPTION
**Pull request by nivanov-ati**
Added AsyncFunctionAction. It's just like FunctionAction, but Async

### CODE REVIEW

**Review by jbcpollak — CHANGES_REQUESTED** [review]
needs unit test

File: common/action/async_function_action.go [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+	fnResult core.ResultDetails
+}
+
+func (a *asyncFunctionAction[Blackboard]) Activate(ctx context.Context, bb Blackboard, evt core.Event) core.ResultDetails {
+	return core.InitRunningResult(a.performFunction)
+}
+
+func (a *asyncFunctionAction[Blackboard]) performFunction(_ context.Context, enqueue core.EnqueueFn) error {
+	a.fnResult = a.Params.Func()
```
Comment (jbcpollak) [inline common/action/async_function_action.go:40]: what if the function needs context information and/or BT params? Should we pass context and params in, or should Func() actually be an object with an interface that is constructed but the caller so they can pass any needed services or dependencies into the function?

Comment (jbcpollak) [reply common/action/async_function_action.go:40]: Note - a unit test or usage in a example project would help answer these questions

Comment (nivanov-ati) [reply common/action/async_function_action.go:40]: I agree re: unit test, but the overall structure is very similar to FunctionAction. Generally, I've been declaring function inside of a method with whatever parameters it needs (I forget what it's called - closure?)

**Review by jbcpollak — APPROVED** [review]

**Pull request merged**